src/app/WriteHandler.h

```diff
@@ -198,6 +207,10 @@ class WriteHandler : public Messaging::ExchangeDelegate
     bool mAttributeWriteSuccessful                = false;
     Optional<AttributeAccessToken> mACLCheckCache = NullOptional;
 
+#if CHIP_CONFIG_USE_DATA_MODEL_INTERFACE
+    DataModel::Provider * mDataModelProvider = nullptr;
```

Comment:
Please watch out for RAM increases.

We should think a bit about how to reduce the set of management/provider/etc pointers this thing hold on to.

Reply:
This one is rough ... I moved the bools into a bitflags so instead of 5xbool I am now using a uint8_t, so this should save the 4 bytes for a pointer (and due to moving things around it may save more on padding...).

I agree that we may want to consider what to do with this, maybe make the IME pass in the pointer during the handle and not making the handler maintain its own pointer. However that change seems to be more complex so I would rather defer it.